src/IncludedCommands/Resources/PersistentBans.lua

```diff
@@ -149,7 +149,15 @@ end
 Fetches the banned players.
 --]]
 function PersistentBans:FetchBannedPlayers()
-    self.BannedUsers = self.DataStore:GetAsync("PercistentBans")
+    local Worked,BannedUsers = pcall(function()
+		return self.DataStore:GetAsync("PercistentBans")
+    end)
+    
+	if Worked then
+		self.BannedUsers = BannedUsers or {}
```

Comment:
This is a bit overkill because calls to `PersistentBans:FetchBannedPlayers()` are called and any error prevents additional processing here. The change made on line 157 would be correct, making the original line on 152 `self.BannedUsers = self.DataStore:GetAsync("PercistentBans") or {}`.

Reply:
Shouldn't there at least be a second fallback at line 65 in that case to set `self.BannedUsers = {}` if the datastore failed? That way the rest of the code could still function properly?

Reply:
Something I forget to mention is the implication of a DataStore failure is the result of resetting vs not updating. If a failure happens, the ban list is reset temporarily and basically means that the ban list is useless during that time, while not updating only results in a delayed banning or unbanning of a smaller portion of the list. Making it my requested change means it will only set the list to empty if the data is empty, but not on failure.

Reply:
Right, I did not notice the assignment in the constructor, removed the assignment on line 65.